[vulkan] addmm support non-vulkan inputs

IvanKobzarev opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #39410 [vulkan] Remove unused variables
* #39384 [vulkan] Shaders caching
* #39282 [vulkan] jit passes for vulkan conv2 prepack and fuse with clamp
* #39115 [vulkan] Conv2d with optional clamp
* **#39078 [vulkan] addmm support non-vulkan inputs**

Adding support of non-vulkan inputs for addmm operator:
if it is not on vulkan - converting to it inside operator, 

if we run torchscript pretrained model - weights of linear op will be on CPU, we need this to run mobilenetV2 on Vulkan backend

Differential Revision: [D21962425](https://our.internmc.facebook.com/intern/diff/D21962425)

IvanKobzarev committed:
[vulkan] addmm support non-vulkan inputs

[ghstack-poisoned]

IvanKobzarev committed:
drop batch norm stub as fuse_models Conv2-BatchNorm2 works on "[vulkan] addmm support non-vulkan inputs"


Adding support of non-vulkan inputs for addmm operator:
if it is not on vulkan - converting to it inside operator, 

if we run torchscript pretrained model - weights of linear op will be on CPU, we need this to run mobilenetV2 on Vulkan backend

[ghstack-poisoned]

IvanKobzarev committed:
rebase on "[vulkan] addmm support non-vulkan inputs"


Adding support of non-vulkan inputs for addmm operator:
if it is not on vulkan - converting to it inside operator, 

if we run torchscript pretrained model - weights of linear op will be on CPU, we need this to run mobilenetV2 on Vulkan backend

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[vulkan] addmm support non-vulkan inputs"


Adding support of non-vulkan inputs for addmm operator:
if it is not on vulkan - converting to it inside operator, 

if we run torchscript pretrained model - weights of linear op will be on CPU, we need this to run mobilenetV2 on Vulkan backend

[ghstack-poisoned]

AshkanAliabadi commented on aten/src/ATen/native/vulkan/VulkanAten.cpp:
```diff
@@ -251,9 +251,11 @@ Tensor vulkan_addmm(
+  VulkanTensor t = vtensor_from_vulkan(self.is_vulkan() ? self : self.vulkan());
+  VulkanTensor m1 =
+      vtensor_from_vulkan(mat1.is_vulkan() ? mat1 : mat1.vulkan());
+  VulkanTensor m2 =
```
const is preferred to protect against the inadvertent changes below.

IvanKobzarev committed:
Update on "[vulkan] addmm support non-vulkan inputs"


Adding support of non-vulkan inputs for addmm operator:
if it is not on vulkan - converting to it inside operator, 

if we run torchscript pretrained model - weights of linear op will be on CPU, we need this to run mobilenetV2 on Vulkan backend

Differential Revision: [D21962425](https://our.internmc.facebook.com/intern/diff/D21962425)

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[vulkan] addmm support non-vulkan inputs"


Adding support of non-vulkan inputs for addmm operator:
if it is not on vulkan - converting to it inside operator, 

if we run torchscript pretrained model - weights of linear op will be on CPU, we need this to run mobilenetV2 on Vulkan backend

Differential Revision: [D21962425](https://our.internmc.facebook.com/intern/diff/D21962425)

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[vulkan] addmm support non-vulkan inputs"


Adding support of non-vulkan inputs for addmm operator:
if it is not on vulkan - converting to it inside operator, 

if we run torchscript pretrained model - weights of linear op will be on CPU, we need this to run mobilenetV2 on Vulkan backend

Differential Revision: [D21962425](https://our.internmc.facebook.com/intern/diff/D21962425)

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[vulkan] addmm support non-vulkan inputs"


Adding support of non-vulkan inputs for addmm operator:
if it is not on vulkan - converting to it inside operator, 

if we run torchscript pretrained model - weights of linear op will be on CPU, we need this to run mobilenetV2 on Vulkan backend

Differential Revision: [D21962425](https://our.internmc.facebook.com/intern/diff/D21962425)

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[vulkan] addmm support non-vulkan inputs"


Adding support of non-vulkan inputs for addmm operator:
if it is not on vulkan - converting to it inside operator, 

if we run torchscript pretrained model - weights of linear op will be on CPU, we need this to run mobilenetV2 on Vulkan backend

Differential Revision: [D21962425](https://our.internmc.facebook.com/intern/diff/D21962425)

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[vulkan] addmm support non-vulkan inputs"


Adding support of non-vulkan inputs for addmm operator:
if it is not on vulkan - converting to it inside operator, 

if we run torchscript pretrained model - weights of linear op will be on CPU, we need this to run mobilenetV2 on Vulkan backend

Differential Revision: [D21962425](https://our.internmc.facebook.com/intern/diff/D21962425)

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[vulkan] addmm support non-vulkan inputs"


Adding support of non-vulkan inputs for addmm operator:
if it is not on vulkan - converting to it inside operator, 

if we run torchscript pretrained model - weights of linear op will be on CPU, we need this to run mobilenetV2 on Vulkan backend

Differential Revision: [D21962425](https://our.internmc.facebook.com/intern/diff/D21962425)

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[vulkan] addmm support non-vulkan inputs"


Adding support of non-vulkan inputs for addmm operator:
if it is not on vulkan - converting to it inside operator, 

if we run torchscript pretrained model - weights of linear op will be on CPU, we need this to run mobilenetV2 on Vulkan backend

Differential Revision: [D21962425](https://our.internmc.facebook.com/intern/diff/D21962425)

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[vulkan] addmm support non-vulkan inputs"


Adding support of non-vulkan inputs for addmm operator:
if it is not on vulkan - converting to it inside operator, 

if we run torchscript pretrained model - weights of linear op will be on CPU, we need this to run mobilenetV2 on Vulkan backend

Differential Revision: [D21962425](https://our.internmc.facebook.com/intern/diff/D21962425)

[ghstack-poisoned]
